src/test/java/core/basesyntax/StrategyTest.java

```diff
@@ -0,0 +1,97 @@
+package core.basesyntax;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class StrategyTest {
+  private static final double DELTA = 0.0;
+  private static Map<String, Double> specialEventsDiscounts;
+
+  @BeforeClass
+  public static void setUp() {
+    specialEventsDiscounts = Map.of("Birthday", 33.0,
+        "Black Friday", 45.0,
+        "New Year", 20.0,
+        "Other event", 0.0);
+  }
+
+  @Test
+  public void getDiscountBySpecialEvent_IsOk() {
+    for (String specialEvent : specialEventsDiscounts.keySet()) {
+      DiscountStrategy discountStrategy = new DiscountStrategy();
+      double expectedDiscount = specialEventsDiscounts.get(specialEvent);
+      double actualDiscount = discountStrategy.getDiscountBySpecialEvent(specialEvent);
+      Assert.assertEquals("Expected discount " + expectedDiscount
+              + ",but your discount " + actualDiscount,
+          expectedDiscount, actualDiscount, DELTA);
```

Comment:
We should check the interface implementation we will receive as a result of method execution.
![image](https://user-images.githubusercontent.com/13794896/136505896-c3f07e34-95e3-43c3-8eb9-de16ba71b3f7.png)